components/locale-chooser.js

```diff
@@ -51,68 +51,135 @@ const Item = glamorous.li((props, {colors, mediaQueries}) => ({
   '&:focus, &:hover, &:active': {
     textDecoration: 'none',
     color: colors.white,
+    backgroundColor: colors.primaryMed,
   },
 }))
 
-const localeContent = ({display, Flag = () => null}) =>
-  (<div>
-    <Flag {...svgStyle} /> <span>{display}</span>
+const localeContent = ({display, Flag}) =>
+  (<div aria-hidden="true">
+    {Flag} <span>{display}</span>
   </div>)
 
-const localeItem = ({key, display, Flag}) =>
-  (<Item key={key}>
-    <Link href={localeToHref(key)}>
-      {localeContent({Flag, display})}
-    </Link>
-  </Item>)
-
 class LocaleChooser extends React.Component {
   state = {
     open: false,
+    locales: [],
+    currentLocale: process.env.LOCALE || fallbackLocale,
+  }
+
+  componentDidMount() {
+    document.addEventListener('click', this.click, true)
+    document.addEventListener('keydown', this.keyDown, true)
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState(() => {
+      return {locales: [...supportedLocales, 'help'].map(mapLocale)}
+    })
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.click, true)
+    document.removeEventListener('keydown', this.keyDown, true)
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.open && !prevState.open) {
+      this['link-en'].focus()
+    }
   }
 
-  handleClick() {
+  toggleOpen = () => {
     this.setState(prevState => {
       return {open: !prevState.open}
     })
   }
 
+  click = event => {
+    if (!this.toggle.contains(event.target) && this.state.open) {
+      this.toggleOpen()
+    }
+  }
+
+  keyDown = event => {
+    // Close on escape
+    if (this.state.open && event.keyCode === 27) {
+      this.toggleOpen()
+      this.toggle.focus()
+    }
+  }
+
+  itemHover = event => {
+    event.target.focus()
+  }
+
+  itemBlur = event => {
+    event.target.blur()
+  }
+
   render() {
     return (
       <Wrapper>
-        <Toggle onClick={this.handleClick.bind(this)} isOpen={this.state.open}>
-          {localeContent(mapLocale(process.env.LOCALE))}
+        <Toggle
+          onClick={this.toggleOpen}
+          isOpen={this.state.open}
+          aria-label={content.ariaLabelButton}
+          aria-haspopup="true"
+          aria-owns="locale-selector"
+          aria-expanded={this.state.open ? 'true' : 'false'}
+          innerRef={button => {
+            this.toggle = button
+          }}
+        >
+          {localeContent(mapLocale(this.state.currentLocale))}
         </Toggle>
-        {this.state.open &&
-          <List aria-label="Locale selector">
-            {supportedLocales.map(l => localeItem(mapLocale(l)))}
-            {localeItem(mapLocale('help'))}
-          </List>}
+        <List
+          id="locale-selector"
+          aria-label={content.ariaLabelList}
+          aria-hidden={!this.state.open}
+          open={this.state.open}
+          top={this.props.top}
+        >
+          {this.state.locales.map(({key, display, Flag}) =>
```

Comment:
To be honest I don't know.
I was thinking it would be better to hide the list with some `css`, because this way the DOM isn't exactly being re-rendered.
Based on your answer I was curious to see how others are doing it (jest and webpack), and they are doing it with `css`.

If you want to I can add a check on the `open` state.